### DISCUSSION TITLE
Ignore scroll when modifier is used

### DISCUSSION THREAD

**Discussion by mattbatt** [sezanzeb/input-remapper#862] [Q&A]
**Main problem:** I don't have the right lingo to search for the answer. 
**What I'm trying to accomplish:** If I tap my mouse's Button EXTRA key it does KEY_HOME, If i hold the Button EXTRA + Scroll up and Scroll down it will turn the volume up and down. 
**The Problem** The Volume change works however the scroll wheel is also scrolling the page up and down in the back ground while changing the volume and it does the KEY_HOME thing in thing in the background too. Which wouldn't be a big deal except that KEY_HOME restarts YouTube videos from the beginning. 

Thank you. 

![image](https://github.com/sezanzeb/input-remapper/assets/2854331/af97fdca-dc64-44d0-b3e1-69723ff1317e)

**Comment by sabudum** [comment]
Create your EXTRA+Scroll as normal, then create three extra remappings:

One will only remap the EXTRA key, as such:
set(extra, 1).hold().set(extra, 0)

Then create two remaps, one for each scroll, up and down, as such:
if_eq($extra, 0, wheel(direction, speed), KEY_VOLUME)

Where direction is the scroll direction (up or down), speed is the scroll speed (probably ok as 1), and KEY_VOLUME is either KEY_VOLUMEUP or KEY_VOLUMEDOWN as needed.

This will create the following scenario:
When pressing down EXTRA, a variable will be written with the value 1, this will inform other macros that EXTRA is being held down.
When EXTRA is released, the value will be set to 0 (zero).

When scrolling up or down, the macro will read the value from the variable, if it is 0 (EXTRA is not being held down), it will send a scroll event, however, if the value is 1 (EXTRA is being held down), then it will send a volume up or down event.

**Comment by drkdr** [comment]
What Sadudum said is mostly right.  It's correct for most keys, but the mouse wheel functions differently.
The scroll wheel sends "Wheel Up", "Wheel Hi RES +", and "Wheel Up + Wheel HI RES +" all together in one tick of the wheel.

SCROLL WHEEL UP:
Click Add, click Record, move the wheel up, click Advanced, under "Event Specific", click "Remove this input" for Wheel Up 1
Click Add, click Record, move the wheel up, click Advanced, under "Event Specific", click "Remove this input" for Wheel HI RES + 1
Click Add, click Record, move the wheel up, leave this as both Wheel Up + Wheel HI RES +
For "Wheel Up" and "Wheel HI RES +" in the "What should be written" field box, type: disable
For "Wheel Up + Wheel HI RES +" type: if_eq($extra, 0, wheel(up, 40), KEY_VOLUMEUP)
For my mouse, a value of 40 was my default scroll speed. Adjust it to your desired speed.

SCROLL WHEEL DOWN:
Do the same thing as SCROLL WHEEL UP, but for Wheel Down

EXTRA KEY:
set(extra, 1).hold_keys(KEY_HOME).set(extra, 0)

**Comment by danicc097** [comment]
In my case what worked for:
```
input-remapper 2.1.1 6b6c19be138992470e2122be43d1227145838520 https://github.com/sezanzeb/input-remapper
python-evdev 1.9.0
```
 to have EXTRA+wheel for volume control was:

Button EXTRA `set(extra, 1).hold().set(extra, 0)`
Button EXTRA + Wheel Down + Wheel HI RES - `if_eq($extra, 0, None, key(KEY_VOLUMEDOWN))`
Button EXTRA + Wheel Up + Wheel HI RES + `if_eq($extra, 0, None, key(KEY_VOLUMEUP))`
Wheel Down (not HI RES) `if_eq($extra, 0, wheel(down, 100))` release input disabled
Wheel Up (not HI RES) `if_eq($extra, 0, wheel(up, 100))` release input disabled

and playing with wheel speeds. both volume control and scrolling are smooth and found no issues thus far.

One thing I haven't figured out is how to keep the default EXTRA key behavior when held and released  without modifiers.
Edit:
```
for plain Button EXTRA: set(extra, 1).set(extra_used_as_modifier, 0).hold().set(extra, 0).if_eq($extra_used_as_modifier, 0, key(BTN_EXTRA), None)
and for extra+wheel up and down: 
if_eq($extra, 0, None, parallel(key(KEY_VOLUMEDOWN), set(extra_used_as_modifier, 1)))
```
seems to work perfectly
